tools/oeedger8r/Emitter.ml

```diff
@@ -217,102 +231,143 @@ let oe_get_param_size (ptype, decl, argstruct) =
 (** Generate the prototype for a given function. Optionally add an
     [oe_enclave_t*] first parameter. *)
 let oe_gen_prototype (fd : func_decl) =
-  let params_str =
-    if List.length fd.plist = 0 then "void" else get_plist_str fd
+  let get_plist_str (plist : pdecl list) =
+    if List.length plist = 0 then "void"
```

Comment:
You know, I noticed some funny stuff with checking `Void`, I'd punted it for a later fix.

Reply:
Oh, no, this is not equivalent to `fd.rtype = Void`, as it is not the return type. This is checking the function takes zero arguments (plist), and if so, generates the correct *C* (not C++) code to say `foo(void)` instead of `foo()`.